sdl-core/src/main/scala/io/smartdatalake/util/hdfs/UCFileSystemFactory.scala

```diff
@@ -72,10 +72,13 @@ object UCFileSystemFactory {
  */
 private[smartdatalake] class DbUtilsInterface(fsUtilsInst: Any, credentialScopeHelperClass: Class[_]) extends SmartDataLakeLogger {
 
-  def checkPermissionAccess[X](pathAndActions: Seq[_], withUnityCatalog: java.lang.Boolean, code : => X): X = {
-    checkPermissionMethod.invoke(fsUtilsInst, pathAndActions, withUnityCatalog, () => code).asInstanceOf[X]
-  }
-  private lazy val checkPermissionMethod = getMethod(fsUtilsInst.getClass, "checkPermission", Seq(classOf[Seq[_]], classOf[Boolean], classOf[Function0[_]]))
+  //As of 14.04.2024, Databricks provides the checkPermission method with two different signatures depenending on where the clusters are hosted.
+  private lazy val checkPermissionMethod = try {
+      getMethod(fsUtilsInst.getClass, "checkPermission", Seq(classOf[Seq[_]], classOf[Boolean], classOf[Function0[_]]))
+    }
+    catch {
+      case nsm: NoSuchMethodException => getMethod(fsUtilsInst.getClass, "checkPermission", Seq(classOf[Seq[_]], classOf[Boolean], classOf[Option[_]], classOf[Function0[_]]));
+    }
   checkPermissionMethod.setAccessible(true)
```

Comment:
I think we could remove this completely, as it is not used currently. I added checkPermissionAccess here intially, as it was also in the stacktrace of accessing hadoop paths, but was not used in the end. registerPathAccess is sufficient, and removing checkPermissionAccess would make it more stable ;-)

Reply:
Method entirely removed